### DISCUSSION TITLE
[SYCL][CUDA] how can I display the kernel name in nvprof

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#5462] [General]
Profiling a SYCL program using Nvidia Profiler shows the following names. How can I make nvprof display the kernel name specified in a SYCL program ?  Thanks.

Typeinfo name for main::[lambda(cl::sycl::handler &) (instance 4)]::operator ()(cl::sycl::handler &…
Typeinfo name for main::[lambda(cl::sycl::handler &) (instance 5)]::operator ()(cl::sycl::handler &…
Typeinfo name for main::[lambda(cl::sycl::handler &) (instance 6)]::operator ()(cl::sycl::handler &…

**Comment by alexbatashev** [comment]
You don't. nvprof is not aware of SYCL, and what you see is the actual kernel name, which is generated by the compiler.

**Reply by zjin-lcf** [reply]
I wonder if people can see the "kernel name" instead of "..." near the end of each line.

**Reply by alexbatashev** [reply]
On SYCL level we have no control over what nvprof shows to the users

**Reply by keryell** [reply]
I guess you can do the same trick that we do to make the name edible by the Xilinx backend by rewriting the name in the compiler or adding a new user-facing name in the inclusion headers so you can decouple the global kernel name from a back-end specific name.
In that case you could rewrite the kernel name to something showable by CUDA environment while working the same with other backends.

**Reply by bader** [reply]
How can we make sure that rewritten name doesn't conflict with other names?

**Reply by zjin-lcf** [reply]
While I don't understand the trick, people may profile SYCL program with nvprof by specifying the kernel name.  Thank you for your answers.

**Reply by AerialMantis** [reply]
So we have ran into this problem when using nsys or ncu (I assume this will be similar to when using nprof) and one work-around to this is to add a wrapper for the kernel invocations which places the lambda in a named function object or namespace, this can improve the name which is propagated to the profilers, though it does require code changes.
